src/nmea.cpp

```diff
@@ -0,0 +1,1048 @@
+		The Talker ID ($--) will vary depending on the satellite system used for the position solution:
+		$GNRMC,092721.00,A,2926.688113,N,11127.771644,E,0.780,,200520,,,D,V*1D
+
+		GPRMC message fields
+		Field	Meaning
+		0	Message ID $GPRMC
+		1	UTC of position fix
+		2	Status A=active or V=void
+		3	Latitude
+		4	Longitude
+		5	Speed over the ground in knots
+		6	Track angle in degrees (True)
+		7	Date
+		8	Magnetic variation in degrees
+		9	The checksum data, always begins with *
+		*/
+		double utc_time = 0.0;
+		char Status = 'V';
+		double lat = 0.0, lon = 0.0;
+		float ground_speed_K =0.0;
+		float A_track =0.0;
+		int nmea_date =0 ;
+		float Mag_var = 0.0;
+		char ns = '?', ew = '?';
+		NMEA_UNUSED(Mag_var);
+
+		_gps_position->nmea_protol_detected = true;
+
+		if (bufptr && *(++bufptr) != ',') { utc_time = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { Status = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { lat = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ns = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { lon = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { ew = *(bufptr++); }
+
+		if (bufptr && *(++bufptr) != ',') { ground_speed_K = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { A_track = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { nmea_date = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (bufptr && *(++bufptr) != ',') { Mag_var = strtod(bufptr, &endp); bufptr = endp; }
+
+		if (ns == 'S') {
+			lat = -lat;
+		}
+
+		if (ew == 'W') {
+			lon = -lon;
+		}
+
+		if (Status == 'V') {
+			return 0;
+		}
+
+		float track_rad = static_cast<float>(A_track) * NMEA_PI/ 180.0f;
+		float velocity_ms =static_cast<float>(ground_speed_K) / 1.9438445f;
+		float velocity_north =static_cast<float>(velocity_ms) * cosf(track_rad);
+		float velocity_east  = static_cast<float>(velocity_ms) *sinf(track_rad);
+		int utc_hour = static_cast<int>(utc_time / 10000);
+		int utc_minute = static_cast<int>((utc_time - utc_hour * 10000) / 100);
+		double utc_sec = static_cast<double>(utc_time - utc_hour * 10000 - utc_minute * 100);
+		int nema_day= static_cast<int>(nmea_date / 10000);
+		int nmea_mth = static_cast<int>((nmea_date - nema_day * 10000) / 100);
+		int nmea_year = static_cast<int>(nmea_date - nema_day * 10000 - nmea_mth * 100);
+		/* convert from degrees, minutes and seconds to degrees * 1e7 */
+		_gps_position->lat = static_cast<int>((int(lat * 0.01) + (lat * 0.01 - int(lat * 0.01)) * 100.0 / 60.0) * 10000000);
+		_gps_position->lon = static_cast<int>((int(lon * 0.01) + (lon * 0.01 - int(lon * 0.01)) * 100.0 / 60.0) * 10000000);
+
+		_gps_position->vel_m_s = velocity_ms;
+		_gps_position->vel_n_m_s =velocity_north;
+		_gps_position->vel_e_m_s =velocity_east;
+		_gps_position->cog_rad =track_rad;
```

Comment:
_gps_position->cog_rad = wrap_pi(track_rad);

because track_rad is [0, 2π), while cog_rad  needs to be [-π, π]